### PR TITLE
Migrate zmalloc.c unit tests to new test framework

### PR DESCRIPTION
**Pull request by karthyuom**
This PR migrates all tests related to zmalloc into new test framework as part of the parent issue https://github.com/valkey-io/valkey/issues/428.

### CODE REVIEW

**Comment by karthyuom** [comment]
Same as this https://github.com/valkey-io/valkey/pull/458#issuecomment-2103744784, I didn't remove legacy zmalloc_test due to the conflict. And, I will raise a follow-up PR to remove them together.

@madolson In the meantime, could you please review and merge this PR, thanks.

File: src/unit/test_zmalloc.c [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+int test_zmallocAllocReallocCallocAndFree(int argc, char **argv, int flags) {
+    UNUSED(argc);
+    UNUSED(argv);
+    UNUSED(flags);
+
+    void *ptr, *ptr2;
+
+    ptr = zmalloc(123);
+    printf("Allocated 123 bytes; used: %zu\n", zmalloc_used_memory());
```
Comment (madolson) [inline src/unit/test_zmalloc.c]:
```suggestion
    TEST_PRINT_INFO("Allocated 123 bytes; used: %zu\n", zmalloc_used_memory());
```

File: src/unit/test_zmalloc.c [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    ptr = zrealloc(ptr, 456);
+    printf("Reallocated to 456 bytes; used: %zu\n", zmalloc_used_memory());
```
Comment (madolson) [inline src/unit/test_zmalloc.c]:
```suggestion
    TEST_PRINT_INFO("Reallocated to 456 bytes; used: %zu\n", zmalloc_used_memory());
```

File: src/unit/test_zmalloc.c [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+    ptr2 = zcalloc(123);
+    printf("Callocated 123 bytes; used: %zu\n", zmalloc_used_memory());
```
Comment (madolson) [inline src/unit/test_zmalloc.c]:
```suggestion
    TEST_PRINT_INFO("Callocated 123 bytes; used: %zu\n", zmalloc_used_memory());
```

File: src/unit/test_zmalloc.c [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+    zfree(ptr);
+    zfree(ptr2);
+    printf("Freed pointers; used: %zu\n", zmalloc_used_memory());
```
Comment (madolson) [inline src/unit/test_zmalloc.c]:
```suggestion
    TEST_PRINT_INFO("Freed pointers; used: %zu\n", zmalloc_used_memory());
```

File: src/unit/test_zmalloc.c [hunk 5]
```diff
@@ -0,0 +1,56 @@
+#include "../zmalloc.c"
+#undef UNUSED
```
Comment (madolson) [inline src/unit/test_zmalloc.c]:
```suggestion
```

Should no longer be needed.

File: src/unit/test_zmalloc.c [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+int test_zmallocAllocZeroByteAndFree(int argc, char **argv, int flags) {
+    UNUSED(argc);
+    UNUSED(argv);
+    UNUSED(flags);
+
+    void *ptr;
+
+    ptr = zmalloc(0);
+    printf("Allocated 0 bytes; used: %zu\n", zmalloc_used_memory());
```
Comment (madolson) [inline src/unit/test_zmalloc.c]:
```suggestion
    TEST_PRINT_INFO("Allocated 0 bytes; used: %zu\n", zmalloc_used_memory());
```

File: src/unit/test_zmalloc.c [hunk 7]
```diff
@@ -0,0 +1,56 @@
+#include "../zmalloc.c"
```
Comment (madolson) [inline src/unit/test_zmalloc.c]:
```suggestion
#include "../zmalloc.h"
```

In this case it looks we're just calling public interfaces, so I think we can just use zmalloc.h

File: src/unit/test_zmalloc.c [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+#undef UNUSED
+#include "test_help.h"
+
+int test_zmallocInitialUsedMemory(int argc, char **argv, int flags) {
+    UNUSED(argc);
+    UNUSED(argv);
+    UNUSED(flags);
+
+    printf("Malloc prefix size: %d\n", (int) PREFIX_SIZE);
```
Comment (madolson) [inline src/unit/test_zmalloc.c]:
```suggestion
    TEST_PRINT_INFO("Malloc prefix size: %d\n", (int) PREFIX_SIZE);
```
I think we want to in the future suppress the output from passing tests, so would rather wrap it in the test print function.

**Review by madolson — COMMENTED** [review]
Mostly lgtm.

**Comment by karthyuom** [comment]
Please refer the PR https://github.com/valkey-io/valkey/pull/493.

**Pull request closed without merging**